### PR TITLE
M3-184 Show user feedback immediately upon Linode power action

### PR DESCRIPTION
**Pull request by asauber**
Transition the state of the UI immediately upon a power action by injecting a fake event into the event stream.

### CODE REVIEW

File: src/features/linodes/LinodesLanding/powerActions.ts [hunk 1]
```diff
@@ -1,12 +1,55 @@
...
+
+export const powerOffLinode = (linode: Linode.Linode) => {
+  Axios.post(`${API_ROOT}/linode/instances/${linode.id}/shutdown`)
+  .then((response) => {
+    linodeEvents$.next(genEvent('linode_shutdown', linode.id, linode.label));
+    resetEventsPolling();
+  });
+};
+
+export const powerOnLinode = (linode: Linode.Linode) => {
```
Comment (rjstires) [inline src/features/linodes/LinodesLanding/powerActions.ts]: In the quest to only send in the data required, would it be best to send in only the Linode ID and label?

File: src/features/linodes/LinodesLanding/powerActions.ts [hunk 2]
```diff
@@ -1,12 +1,55 @@
...
+    id: 1, /* NB: fake but valid Event ID */
+    percent_complete: 0,
+    seen: false,
+    entity: {
+      url: `/v4/linode/instances/${linodeID}`,
+      label: `${linodeLabel}`,
+      type: 'linode',
+      id: linodeID,
+    },
+  } as Linode.Event;
```
Comment (rjstires) [inline src/features/linodes/LinodesLanding/powerActions.ts]: We can use `Partial<Linode.Event>` so we don't have to stub a bunch of data here.

Comment (asauber) [reply src/features/linodes/LinodesLanding/powerActions.ts]: Partial<Linode.Event> yields a type, which wouldn't really be useful here

File: src/features/linodes/LinodesLanding/powerActions.ts (hunk 1 above)
Comment (asauber) [reply src/features/linodes/LinodesLanding/powerActions.ts]: yes

**Pull request merged**